main.go

```diff
@@ -96,7 +182,7 @@ func Run(name string, arg ...string) {
 
 // Error is the command's error, if any.
 func (c *Cmd) Error() error {
-	c.validate()
+	c.validateIsDone()
```

Comment:
I'm having second thoughts about these validation methods, I'm no longer sure it's a good idea to halt the execution (`log.Fatal()`), it might be better to return an error and let the caller decide what to do. For example:
```go
if !c.hasFinished() {
	log.Printf("The program hasn't started or is still running...")
	// Should I halt the program? Or should I return a special error?
}
````

Reply:
Yes, I think `log.Fatal` is not a great idea. Did some tests and when this is called it stops the rest of the test suite from running. Returning an error could work, but would make tests a bit more verbose. Would it be an idea to pass `*testing.T` down to the command so we can replace the `log.Fatal` calls with `t.Fatal`?